### PR TITLE
New style softplus function

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
Please see #3147.

### CODE REVIEW

**Comment by keisuke-umezawa** [comment]
Some errors except python3.5
```
======================================================================
ERROR: Failure: ImportError (cannot import name configuration)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Miniconda-x64\lib\site-packages\nose-1.3.7-py2.7.egg\nose\loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "C:\Miniconda-x64\lib\site-packages\nose-1.3.7-py2.7.egg\nose\importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "C:\Miniconda-x64\lib\site-packages\nose-1.3.7-py2.7.egg\nose\importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "C:\projects\chainer\tests\chainer_tests\utils_tests\test_walker_alias.py", line 5, in <module>
    from chainer import cuda
  File "C:\Miniconda-x64\lib\site-packages\chainer-3.0.0rc1-py2.7.egg\chainer\__init__.py", line 8, in <module>
    from chainer import configuration  # NOQA
ImportError: cannot import name configuration
----------------------------------------------------------------------
```

File: chainer/functions/activation/softplus.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import numpy
 
 from chainer import cuda
-from chainer import function
+from chainer import function_node
+from chainer import functions
```
Comment (okuta) [inline chainer/functions/activation/softplus.py]: Please use `import chainer.functions` .

Comment (keisuke-umezawa) [reply chainer/functions/activation/softplus.py]: @okuta Thank you for the advice. I fixed it.

File: chainer/functions/activation/softplus.py [hunk 2]
```diff
@@ -40,22 +43,49 @@ def forward_gpu(self, inputs):
...
+            'softplus_bwd')(x, gy, self.beta)
         return gx,
 
+    def backward(self, indexes, grad_outputs):
+        x, gy = self.get_retained_inputs()
+        g, = grad_outputs
+        e = chainer.functions.exp(self.beta * x)
+        ggx = g * gy * self.beta * e / (1 + e) ** 2
+        ggy = SoftplusGrad((self.beta,)).apply((x, g))[0]
+        return (ggx, ggy)
```
Comment (okuta) [inline chainer/functions/activation/softplus.py]: Please remove this parenthesis.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by keisuke-umezawa** [comment]
Thank you for the review. I fixed it.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**